### ISSUE TITLE
unable to include into ionic project 

### ISSUE THREAD

**Issue by tanzeelrana** [rh389/dynamodb-geo.js#3]
`npm install dynamodb-geo --save`

then when I do a `ionic serve` I get the following errors : 

![image](https://cloud.githubusercontent.com/assets/2520028/26524426/eeaa13dc-42ff-11e7-8811-46377ed2c506.png)

![image](https://cloud.githubusercontent.com/assets/2520028/26524428/f5522dfa-42ff-11e7-8328-34f6804f1d6b.png)

![image](https://cloud.githubusercontent.com/assets/2520028/26524429/fe162a9a-42ff-11e7-83df-7d8d02645b21.png)

My local env details: 

`Ionic Framework: 3.1.1`
`Ionic App Scripts: 1.3.7`
`Angular Core: 4.0.2`
`Angular Compiler CLI: 4.0.2`
`Node: 6.7.0`
`OS Platform: Linux 4.4`
`Navigator Platform: Linux x86_64`
`User Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)`
`Chrome/58.0.3029.110 Safari/537.36`

**Comment by rh389** [comment]
Looks like there's a conflict between this library's dependency on `@types/es6-shim` and `@types/core-js` which I guess you have installed (Ionic recommends it).

I'll have to do some testing to work out the best solution to this. In the meantime, you could try deleting the `@types/es6-shim` directory from your `node_modules`.

**Comment by tanzeelrana** [comment]
I got it working, you were right I had some conflicting libraries. Took me a good few hours to figure this out. 

my tsconfig.json : 

```
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "lib": [
      "dom",
      "es2015"
    ],
    "module": "es2015",
    "moduleResolution": "node",
    "sourceMap": true,
    "target": "es5"
  },
  "include": [
    "src/**/*.ts"
  ],
  "exclude": [
    "node_modules"
  ],
  "compileOnSave": false,
  "atom": {
    "rewriteTsconfig": false
  }
}
``` 

I also had to install the following : 

```
"@types/long": "^3.0.31",
"@types/decimal.js": "0.0.21-alpha",
"@types/node": "^7.0.22",
"angular2-uuid": "^1.1.1",
"decimal": "0.0.2",
"nodes2ts": "^1.1.9",
```

I am still not sure which `decimal` module to use but for the moment my project is working without any errors. Could this be an issue later on somehow? 

Thanks for the reply I will close the ticket for now

**Comment by rh389** [comment]
You shouldn't need to install anything else explicitly - those packages are mostly part of this project's dependency tree so npm will automatically install them (and `nodes2ts` requires `decimal.js`, not `decimal` - but again you shouldn't need to install either explicitly unless you use them directly elsewhere).

I've published 0.2.2 which doesn't depend on `@types/es6-shim` (instead using typescript's `es2015` lib). That should work better "out of the box" for projects that already include a different definition of ES6 types.
